mbuild/compound.py

```diff
@@ -312,8 +323,205 @@ def particles_by_name(self, name):
             if particle.name == name:
                 yield particle
 
+    @property
+    def contains_rigid(self):
+        """Returns True if the Compound contains rigid bodies
+
+        If the Compound contains any particle with a rigid_id != None 
+        then contains_rigid will return True. If the Compound has no
+        children (i.e. the Compound resides at the bottom of the containment
+        hierarchy) then contains_rigid will return False.
+
+        Returns
+        -------
+        bool
+            True if the Compound contains any particle with a rigid_id != None
+
+        """
+        if any(particle.rigid_id is not None for particle in self._particles()):
+            return True
+        else:
+            return False
+
+    @property
+    def max_rigid_id(self):
+        """Returns the maximum rigid body ID contained in the Compound.
+
+        This is usually used by compound.root to determine the maximum
+        rigid_id in the containment hierarchy.
+
+        Returns
+        -------
+        int or None
+            The maximum rigid body ID contained in the Compound. If no
+            rigid body IDs are found, None is returned
+
+        """
+        if self.contains_rigid:
+            return max([particle.rigid_id for particle in self.particles()
+                        if particle.rigid_id is not None])
+        else:
+            return None
+
+    def rigid_particles(self, rigid_id=None):
+        """Generate all particles in rigid bodies.
+
+        If a rigid_id is specified, then this function will only yield particles
+        with a matching rigid_id.
+
+        Parameters
+        ----------
+        rigid_id : int, optional
+            Include only particles with this rigid body ID
+
+        Yields
+        ------
+        mb.Compound
+            The next particle with a rigid_id that is not None, or the next
+            particle with a matching rigid_id if specified
+
+        """
+        for particle in self.particles():
+            if rigid_id is not None:
+                if particle.rigid_id == rigid_id:
+                    yield particle
+            else:
+                if particle.rigid_id is not None:
+                    yield particle
+
+    def label_rigid_bodies(self, discrete_bodies=None, rigid_particles=None):
+        """Designate which Compounds should be treated as rigid bodies
+
+        If no arguments are provided, this function will treat the compound
+        as a single rigid body by providing all particles in `self` with the
+        same rigid_id. If `discrete_bodies` is not None, each instance of 
+        a Compound with a name found in `discrete_bodies` will be treated as a 
+        unique rigid body. If `rigid_particles` is not None, only Compounds at
```

Comment:
Let's keep using the terminology for Particles vs Compound: 

`...only Particles (Compounds at the bottom of the hierarchy)...`